packages/runtime-core/src/componentEmits.ts

```diff
@@ -189,6 +190,8 @@ export function emit(
     compatModelEmit(instance, event, args)
     return compatInstanceEmit(instance, event, args)
   }
+  
+  return result
```

Comment:
Should change type declaration too
https://github.com/vuejs/core/blob/45782df349fbfd88e9442ed76fa2b108ea5bf3a6/packages/runtime-core/src/componentEmits.ts#L57-L70